### PR TITLE
Add auto translation to `Tree`

### LINKED ISSUE #80549: Some Control Type not auto translate when enable auto_translate
### Godot version

v4.2.dev2.official [da81ca62a]

### System information

window10 64bit

### Issue description

I'm create a translate file like this:
image

And coding like this:
image

And enable all Control's `Auto Translate` property:
image

Then I found some Control Type (Tree, GraphNode) not translated.
image

### Steps to reproduce

/

### Minimal reproduction project

[NewTest.zip](https://github.com/godotengine/godot/files/12327503/NewTest.zip)

### PR DESCRIPTION
**Pull request by YeldhamDev**
Should fully close #80549.

### CODE REVIEW

File: scene/gui/tree.cpp [hunk 1]
```diff
@@ -315,6 +315,9 @@ void TreeItem::set_text(int p_column, String p_text) {
 			cells.write[p_column].max = MAX(cells[p_column].max, value);
 		}
 		cells.write[p_column].step = 0;
+	} else {
+		// Don't auto translate if it's in string mode and editable, as the text can be changed to anything by the player.
+		cells.write[p_column].xl_text = tree && (!cells[p_column].editable || cells[p_column].mode != TreeItem::CELL_MODE_STRING) ? tree->atr(p_text) : p_text;
```
Comment (akien-mga) [inline scene/gui/tree.cpp:320]: Chaining an assignment with a second statement like this looks pretty weird and hard to read.

Comment (akien-mga) [reply scene/gui/tree.cpp:320]: Oh well I misread (on phone), it's a convoluted ternary assignment. I would add an extra pair of parentheses around the full condition, or make this a plain `if () {} else {}`.

File: scene/gui/tree.cpp [hunk 2]
```diff
@@ -1966,11 +1970,17 @@ void Tree::update_item_cell(TreeItem *p_item, int p_col) {
 			valtext = String::num(p_item->cells[p_col].val, Math::range_step_decimals(p_item->cells[p_col].step));
 		}
 	} else {
-		valtext = p_item->cells[p_col].text;
+		// Don't auto translate if it's in string mode and editable, as the text can be changed to anything by the player.
```
Comment (KoBeWi) [inline scene/gui/tree.cpp:1973]:
```suggestion
		// Don't auto translate if it's in string mode and editable, as the text can be changed to anything by the user.
```
I think it's more accurate in GUI context.

**Review by KoBeWi — APPROVED** [review]
I didn't test every cell mode, but the base functionality is working correctly.
I was surprised that it doesn't cause any unwanted translations in the editor, even with non-editable Trees 🤔

**Comment by YeldhamDev** [comment]
Changes made.

**Comment by akien-mga** [comment]
Thanks!

**Comment by KoBeWi** [comment]
> I was surprised that it doesn't cause any unwanted translations in the editor, even with non-editable Trees

Apparently it auto-translates file names in FileSystem.

https://github.com/godotengine/godot/assets/2223172/87c1a3de-b302-4d70-9bea-730c753a1e95

**Pull request merged**